src/prefect/deployments/runner.py

```diff
@@ -351,8 +357,32 @@ async def apply(
                 trigger.set_deployment_id(deployment_id)
                 await client.create_automation(trigger.as_automation())
 
+            # We plan to support SLA configuration on the Prefect Server in the future.
+            # For now, we only support it on Prefect Cloud.
+            if self.sla:
+                await self._create_slas(deployment_id, client)
+
             return deployment_id
 
+    async def _create_slas(self, deployment_id: UUID, client: PrefectClient):
+        if not isinstance(self.sla, list):
+            self.sla = [self.sla]
+
+        if client.server_type == ServerType.CLOUD:
+            exceptions = []
+            for sla in self.sla:
+                try:
+                    sla.set_deployment_id(deployment_id)
+                    await client.create_sla(sla)
+                except Exception as e:
+                    exceptions.append((f"SLA named '{sla.name}' failed to create", e))
+            if exceptions:
+                raise Exception(exceptions)
```

Comment:
This looks like a good place to raise an [`ExceptionGroup`](https://docs.python.org/3/tutorial/errors.html#tut-exception-groups). `ExceptionGroup` was added in 3.11, but you can import a backported version from `exceptiongroup`.

Reply:
Great call, will do!